src/App.tsx

```diff
@@ -15,36 +16,89 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  default = '',
+  alphabetically = 'alphabetically',
+  length = 'length',
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.default);
+  const [reverse, setReverse] = useState(false);
+  const preparedGoods = [...goodsFromServer];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.alphabetically:
+          return good1.localeCompare(good2);
+        case SortType.length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  const reset = () => {
+    setReverse(false);
+    setSortField(SortType.default);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortType.alphabetically,
+          })}
+          onClick={() => setSortField(SortType.alphabetically)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SortType.length,
+          })}
+          onClick={() => setSortField(SortType.length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !reverse,
+          })}
+          onClick={() => setReverse(!reverse)}
```

Comment:
```suggestion
          onClick={() => setReverse(isReversed => !isReversed)}
```

Always use callback to update state based on previous value